### ISSUE TITLE
Deprecation notice: Explicitly mark $authException as nullable in JWTAuthenticator::start()

### ISSUE THREAD

**Issue by gzim324** [lexik/LexikJWTAuthenticationBundle#1260]
### Summary

When using `LexikJWTAuthenticationBundle` with PHP 8.4, a deprecation notice is triggered due to the `start()` method in `JWTAuthenticator` not explicitly marking the `$authException` parameter as nullable. This is deprecated in PHP 8.1 and above.

### Deprecation Notice
Lexik\Bundle\JWTAuthenticationBundle\Security\Authenticator\JWTAuthenticator::start(): Implicitly marking parameter $authException as nullable is deprecated, the explicit nullable type must be used instead

### Steps to Reproduce

1. Use PHP 8.4
2. Configure `LexikJWTAuthenticationBundle` in a Symfony project.
3. Trigger any functionality that calls the `start()` method (e.g., unauthenticated access to protected routes).
4. Observe the deprecation warning.

### Proposed Fix

Update the `start()` method in `JWTAuthenticator` to explicitly declare the `$authException` parameter as nullable.

#### Current Code:
```php
public function start(Request $request, AuthenticationException $authException = null): Response
{
    // ...
}

### Suggested Fix:
```
public function start(Request $request, ?AuthenticationException $authException = null): Response
{
    // ...
}
```

### Environment
PHP Version: 8.4
Symfony Version: 7.2
LexikJWTAuthenticationBundle Version: 3.1.0

**Comment by gzim324** [comment]
I created pull request with solution (ready to merge).
https://github.com/lexik/LexikJWTAuthenticationBundle/pull/1261

Also was removed other depreprecation notice.

**Comment by lxregistry** [comment]
Deprecated:  Lexik\Bundle\JWTAuthenticationBundle\Security\Authenticator\JWTAuthenticator::__construct(): Implicitly marking parameter $translator as nullable is deprecated, the explicit nullable type must be used instead in /var/www/vhosts/domain.local/vendor/lexik/jwt-authentication-bundle/Security/Authenticator/JWTAuthenticator.php on line 46
Deprecated:  Lexik\Bundle\JWTAuthenticationBundle\Security\Authenticator\JWTAuthenticator::start(): Implicitly marking parameter $authException as nullable is deprecated, the explicit nullable type must be used instead in /var/www/vhosts/domain.local/vendor/lexik/jwt-authentication-bundle/Security/Authenticator/JWTAuthenticator.php on line 63
Deprecated:  Lexik\Bundle\JWTAuthenticationBundle\Security\Http\Authentication\AuthenticationFailureHandler::__construct(): Implicitly marking parameter $translator as nullable is deprecated, the explicit nullable type must be used instead in /var/www/vhosts/domain.local/vendor/lexik/jwt-authentication-bundle/Security/Http/Authentication/AuthenticationFailureHandler.php on line 25
Deprecated:  Lexik\Bundle\JWTAuthenticationBundle\Services\JWTManager::__construct(): Implicitly marking parameter $payloadEnrichment as nullable is deprecated, the explicit nullable type must be used instead in /var/www/vhosts/domain.local/vendor/lexik/jwt-authentication-bundle/Services/JWTManager.php on line 33
Deprecated:  Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProvider\LcobucciJWSProvider::__construct(): Implicitly marking parameter $clock as nullable is deprecated, the explicit nullable type must be used instead in /var/www/vhosts/domain.local/vendor/lexik/jwt-authentication-bundle/Services/JWSProvider/LcobucciJWSProvider.php on line 47

**Comment by chalasr** [comment]
fixed in #1238
